packages/ag-charts-community/src/chart/axis/axisTickGenerator.ts

```diff
@@ -208,13 +222,31 @@ export class AxisTickGenerator<S extends Scale<D, number, TickInterval<S>>, D> {
         }
 
         textAlign = getTextAlign(parallel, configuredRotation, autoRotation, sideFlag, regularFlipFlag);
-        const combinedRotation = defaultRotation + configuredRotation + autoRotation;
+        const rotation = configuredRotation + autoRotation;
 
         if (!secondaryAxis && tickData.rawTicks.length > 0) {
             primaryTickCount = tickData.rawTicks.length;
         }
 
-        return { tickData, primaryTickCount, combinedRotation, textBaseline, textAlign };
+        if (removeOverflowingLabels && tickData.ticks.length > 2) {
+            const labelData = getLabelData(tickData, autoRotation);
+            const lastTick = tickData.ticks.at(-1);
+            const lastLabel = labelData.at(-1);
+            if (
+                lastTick != null &&
+                lastLabel != null &&
+                lastTick.translationY + lastLabel.label.width / 2 > this.axis.range[1]
```

Comment:
This check is missing the padding from the original code, which is why labels are being removed although not clipped